### ISSUE TITLE
Unexpected behavior in Apps Jobs processors

### ISSUE THREAD

**Issue by felipetomm** [RocketChat/Rocket.Chat#22863]
### Description:

The `processor` that run in Apps is running in a loop, independent of your time to run new job.

### Steps to reproduce:

1. Create an App with an simple OneTime job with an time to run again;
2. In Apps log screen, see growing the logs list, without respect the interval time.

### Expected behavior:

When an job have an interval time, the expected behavior is run the job only when the time is going.

### Actual behavior:

In a job with interval time, after first execution the processor run after every time, like a loop.

### Server Setup Information:

- Version of Rocket.Chat Server: 3.17.0 Enterprise
- Operating System: Linux
- Deployment Method: docker
- Number of Running Instances: 3
- DB Replicaset Oplog: 
- NodeJS Version: 12.22.1
- MongoDB Version: 4.0.25

### Client Setup Information

- Desktop App or Browser Version:
- Operating System:

#### Example of Job on App
```ts
public async extendConfiguration(configuration: IConfigurationExtend): Promise<void> {
        await configuration.scheduler.registerProcessors([
            {
                id: 'SIMPLE_JOB',
                startupSetting: {
                    type: StartupType.ONETIME,
                    when: new Date( new Date().getTime() + 1 * 60000 )
                },
                processor: async(jobContext: IJobContext,read: IRead, modify: IModify, http: IHttp, persis: IPersistence ) => {
                    this.getLogger().log('Simple job running')
                }
            }
        ])
    }
```

#### Example of the behavior

![Peek 04-08-2021 17-32-JobBug](https://user-images.githubusercontent.com/13468070/128251031-4ff6673f-ff22-4d13-8621-4f77b0ed9220.gif)

**Comment by johncrisp** [comment]
Hi and thanks for reporting this.

We'll check with the Product Manager as this should probably be in a different repo.

We'll keep you posted.

**Comment by johncrisp** [comment]
Hi, I spoke the Product team who replied as follows:

We are aware about it. It's not a problem with Apps Engine itself, but the Apps Engine squad are taking care of it.

This PR should fix it.

https://github.com/RocketChat/Rocket.Chat/pull/22882

Thanks.

**Comment by felipetomm** [comment]
Thanks @johncrisp for your report!
Has you describe, the problem is solved in [3.17.1](https://github.com/RocketChat/Rocket.Chat/releases/tag/3.17.1) :)
